client/state/posts/actions.js

```diff
@@ -774,3 +775,17 @@ export const autosave = () => async ( dispatch, getState ) => {
 
 	return await dispatch( saveEdited( { recordSaveEvent: false, autosave: true } ) );
 };
+
+/**
+ * Save a revision of the passed-in post. Calls `savePost` which initiates a network call.
+ * @param  {int} siteId		The site ID for use in the `savePost` call
+ * @param  {object} post   post-like object with keys: 'ID', 'content', 'excerpt', 'title'
+ * @returns {Promise} The return value from `savePost`
+ */
+export const saveRevision = ( siteId, post ) =>
+	savePost( siteId, null, {
+		...pick( post, POST_REVISION_FIELDS ),
+		parent: post.ID,
+		status: 'inherit',
+		type: 'revision',
+	} );
```

Comment:
The `savePost` action dispatches `POST_SAVE`, `POST_SAVE_SUCCESS` and `POSTS_RECEIVE` while saving the post or revision. That updates the structures in `state.posts`. And the revision doesn't really belong there.

It doesn't immediately break anything, just adds more entropy to already complex and fragile code.

It would be better to directly call `wpcom.site().post().add()` in the `saveRevision` action thunk, without the extra dispatches.